test/common/wait.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"bufio"
+	"io"
+	"strings"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// WaitForObject waits for object to match the desired status.
+func WaitForObject(tl TestLogger, namespace, name string, objectGetter func(namespace, name string) (pkgruntime.Object, error), desired pkgruntime.Object, interval, timeout time.Duration, equivalent func(actual, desired pkgruntime.Object) bool) {
+	var actual pkgruntime.Object
+	err := wait.PollImmediate(interval, timeout, func() (exist bool, err error) {
+		actual, err = objectGetter(namespace, name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+
+		return equivalent(actual, desired), nil
+	})
+	if err != nil {
+		tl.Fatalf("Timedout waiting for desired state, \ndesired: %#v\nactual:  %#v", desired, actual)
+	}
+}
+
+// WaitUntilLogStreamContains waits for the given stream to contain the
+// substring until the end of the stream or timeout.
+func WaitUntilLogStreamContains(tl TestLogger, stream io.ReadCloser, substr string, timeout time.Duration) bool {
```

Comment:
Why is this method accepting a timeout if every caller is passing the same value?

Edit: I see it's related to the source of the timeout being the framework package.  Hopefully moving this file to the framework package would allow use of the context value rather than having to pass it in.